### PR TITLE
staking: Add a burn address

### PR DESCRIPTION
**Pull request by Yawning**
Yet another PR that can rot for months in "don't want to fork the 22.x branch yet" purgatory.

* [x] Define a burn address
* [x] Support burning based off xfers to the burn address
* [x] Test cases
* ~Migration~

Implements #4755

### CODE REVIEW

**Comment by kostko** [comment]
Let's rebase now that master can break consensus.

**Comment by Yawning** [comment]
I'm not sure what sort of migration is required.  The genesis doc sanity check in my branch checks that the burn address is empty...

**Review by kostko — APPROVED** [review]

File: go/staking/api/address_internal.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	)
+
+	// BurnAddress is the burn address.  Transfers sent to this address
+	// are treated identically to token burn by the transfer originator.
+	//
+	// This address is reserved to prevent it beign accidentally used in
+	// the actual ledger.
+	//
+	// oasis1qzq8u7xs328puu2jy524w3fygzs63rv3u5967970
+	BurnAddress = func() Address {
```
Comment (peternose) [inline go/staking/api/address_internal.go:38]: Could also put address generation procedure (this code) in the comment and tests and hardcore the address.

Comment (Yawning) [reply go/staking/api/address_internal.go:38]: Eh, it's cheap and I added tests for the addresess (`address_test.go:TestInternal`).

**Pull request merged**